test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -59,30 +60,41 @@ public string[] GetInstallCommandsFromLocal(string path)
             };
         }
 
-        public string[] GetInstallCommandsFromMicrosoftProd() => this.packageExtension switch
+        public string[] GetInstallCommandsFromMicrosoftProd(Option<Uri> proxy)
         {
-            SupportedPackageExtension.Deb => new[]
+            var curl = "curl";
+            var prefix = string.Empty;
+            proxy.ForEach(url =>
             {
-                // Based on instructions at:
-                // https://github.com/MicrosoftDocs/azure-docs/blob/058084949656b7df518b64bfc5728402c730536a/articles/iot-edge/how-to-install-iot-edge-linux.md
-                // TODO: 8/30/2019 support curl behind a proxy
-                $"curl https://packages.microsoft.com/config/{this.os}/{this.version}/multiarch/prod.list > /etc/apt/sources.list.d/microsoft-prod.list",
-                "curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg",
-                $"apt-get update",
-                $"apt-get install --yes iotedge"
-            },
-            SupportedPackageExtension.Rpm => new[]
+                curl += $" -x {url}";
+                prefix = $"http_proxy={url} https_proxy={url} ";
+            });
+
+            return this.packageExtension switch
             {
-                $"rpm -iv --replacepkgs https://packages.microsoft.com/config/{this.os}/{this.version}/packages-microsoft-prod.rpm",
-                $"yum updateinfo",
-                $"yum install --yes iotedge",
-                "pathToSystemdConfig=$(systemctl cat iotedge | head -n 1)",
-                "sed 's/=on-failure/=no/g' ${pathToSystemdConfig#?} > ~/override.conf",
-                "sudo mv -f ~/override.conf ${pathToSystemdConfig#?}",
-                "sudo systemctl daemon-reload"
-            },
-            _ => throw new NotImplementedException($"Don't know how to install daemon on for '.{this.packageExtension}'"),
-        };
+                SupportedPackageExtension.Deb => new[]
+                {
+                    // Based on instructions at:
+                    // https://github.com/MicrosoftDocs/azure-docs/blob/058084949656b7df518b64bfc5728402c730536a/articles/iot-edge/how-to-install-iot-edge-linux.md
+                    $"{curl} https://packages.microsoft.com/config/{this.os}/{this.version}/multiarch/prod.list > /etc/apt/sources.list.d/microsoft-prod.list",
+                    $"{curl} https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > /etc/apt/trusted.gpg.d/microsoft.gpg",
+                    $"{prefix}apt-get update",
+                    $"{prefix}apt-get install --yes iotedge"
+                },
+                // TODO: 10/14/2020 support yum behind a proxy
+                SupportedPackageExtension.Rpm => new[]
```

Comment:
What's the existing issue with yum behind proxy?

Reply:
None that I know of, but we don't run CentOs behind a proxy in our test pipeline so I didn't implement it.

Reply:
I'll replace the TODO with a comment clarifying that we haven't implemented proxy support in the CentOs path.

Reply:
Updated